music_assistant/providers/spotify/provider.py

```diff
@@ -267,6 +330,107 @@ async def get_playlist(self, prov_playlist_id: str) -> Playlist:
         playlist_obj = await self._get_data(f"playlists/{prov_playlist_id}")
         return parse_playlist(playlist_obj, self)
 
+    # Podcast methods (integrated from podcast_helpers.py)
+    async def get_podcast(self, prov_podcast_id: str) -> Podcast:
+        """Get full podcast details by id."""
+        podcast_obj = await self._get_data(f"shows/{prov_podcast_id}")
+        if not podcast_obj:
+            raise MediaNotFoundError(f"Podcast not found: {prov_podcast_id}")
+        return parse_podcast(podcast_obj, self)
+
+    async def get_podcast_episodes(
+        self, prov_podcast_id: str
+    ) -> AsyncGenerator[PodcastEpisode, None]:
+        """Get all podcast episodes."""
+        podcast = await self.get_podcast(prov_podcast_id)
+        episode_position = 1
+
+        async for item in self._get_all_items(
+            f"shows/{prov_podcast_id}/episodes", market="from_token"
+        ):
+            if not (item and item["id"]):
+                continue
+
+            episode = parse_podcast_episode(item, self, podcast=podcast)
+            episode.position = episode_position
+            episode_position += 1
+            yield episode
+
+    async def get_podcast_episode(self, prov_episode_id: str) -> PodcastEpisode:
+        """Get full podcast episode details by id."""
+        episode_obj = await self._get_data(f"episodes/{prov_episode_id}", market="from_token")
+        if not episode_obj:
+            raise MediaNotFoundError(f"Episode not found: {prov_episode_id}")
+        return parse_podcast_episode(episode_obj, self)
+
+    async def get_resume_position(self, item_id: str, media_type: MediaType) -> tuple[bool, int]:
+        """Get resume position for episode from Spotify."""
+        if media_type != MediaType.PODCAST_EPISODE:
+            raise NotImplementedError("Resume position only supported for podcast episodes")
```

Comment:
can we at least throw an exception here based on MusicAssistantError ?
Maybe an InvalidDataError ?

Reply:
method reworked generally to provide better exception handling